api/src/main/java/org/openmrs/module/eptsreports/reporting/calculation/dsd/NextAndPrevDatesCalculation.java

```diff
@@ -62,6 +62,8 @@ public CalculationResultMap evaluate(
       while (iterator.hasNext()) {
         Obs obs = iterator.next();
         if (obs.getVoided()
+            || obs.getValueDate() == null
```

Comment:
getValueDate & getValueDateTime return the same object - do we need both?

Reply:
GetValueDateTime has the time component while getValueDate has only the date.

Reply:
They return the same object, see here: https://github.com/openmrs/openmrs-core/blob/06593ec985c2e2064e13d172159b04f4398ca1ae/api/src/main/java/org/openmrs/Obs.java#L804-L823

Reply:
You are right @nthfloor We only need one, either return both date and time components.

Reply:
I will go ahead and make the changes.